op-chain-ops/genesis/genesis.go

```diff
@@ -91,35 +92,22 @@ func NewL2Genesis(config *DeployConfig, l1StartHeader *types.Header) (*core.Gene
 		difficulty = newHexBig(0)
 	}
 
-	extraData := config.L2GenesisBlockExtraData
-	if extraData == nil {
-		// L2GenesisBlockExtraData is optional, so use a default value when nil
-		extraData = BedrockTransitionBlockExtraData
-	}
-	// Ensure that the extradata is valid
-	if size := len(extraData); size > 32 {
-		return nil, fmt.Errorf("transition block extradata too long: %d", size)
-	}
-
 	genesis := &core.Genesis{
-		Config:     &optimismChainConfig,
-		Nonce:      uint64(config.L2GenesisBlockNonce),
-		Timestamp:  l1StartTime,
-		ExtraData:  extraData,
-		GasLimit:   uint64(gasLimit),
-		Difficulty: difficulty.ToInt(),
-		Mixhash:    config.L2GenesisBlockMixHash,
-		Coinbase:   predeploys.SequencerFeeVaultAddr,
-		Number:     uint64(config.L2GenesisBlockNumber),
-		GasUsed:    uint64(config.L2GenesisBlockGasUsed),
-		ParentHash: config.L2GenesisBlockParentHash,
-		BaseFee:    baseFee.ToInt(),
-		Alloc:      map[common.Address]types.Account{},
-	}
-
-	if optimismChainConfig.IsEcotone(genesis.Timestamp) {
-		genesis.BlobGasUsed = u64ptr(0)
-		genesis.ExcessBlobGas = u64ptr(0)
+		Config:        &optimismChainConfig,
+		Nonce:         uint64(config.L2GenesisBlockNonce),
+		Timestamp:     l1StartTime,
+		ExtraData:     HoloceneExtraData,
+		GasLimit:      uint64(gasLimit),
+		Difficulty:    difficulty.ToInt(),
+		Mixhash:       config.L2GenesisBlockMixHash,
+		Coinbase:      predeploys.SequencerFeeVaultAddr,
+		Number:        uint64(config.L2GenesisBlockNumber),
+		GasUsed:       uint64(config.L2GenesisBlockGasUsed),
+		ParentHash:    config.L2GenesisBlockParentHash,
+		BaseFee:       baseFee.ToInt(),
+		BlobGasUsed:   u64ptr(0),
+		ExcessBlobGas: u64ptr(0),
```

Comment:
We don't support genesis prior to ecotone anymore?

Reply:
True, we may want to keep around the branching logic that was removed here

Reply:
Ah that makes sense. Restored branching logic with guards for both Ecotone & Holocene.